components/electric/lib/electric/postgres/oid_database.ex

```diff
@@ -13,70 +17,47 @@ defmodule Electric.Postgres.OidDatabase do
     GenServer.call(server, {:save_oids, Enum.map(values, &pg_type_from_tuple/1)})
   end
 
-  @doc """
-  Get an atom name by the type OID
-  """
-  @spec name_for_oid(integer()) :: atom() | {:array, atom()}
+  @spec name_for_oid(oid) :: type_name | {:array, type_name}
   def name_for_oid(oid) do
-    case :ets.lookup(@ets_table_name, oid) do
+    case :ets.lookup(@oid_table, oid) do
       [pg_type(is_array: false, name: name)] -> name
-      [pg_type(is_array: true, element_oid: element)] -> {:array, name_for_oid(element)}
-      _ -> raise("Unknown OID #{oid}, cannot get a name that won't result in data loss")
+      [pg_type(is_array: true, element_oid: element_oid)] -> {:array, name_for_oid(element_oid)}
+      _ -> raise "Unknown OID #{oid}, cannot get a name that won't result in data loss"
     end
   end
 
-  @doc """
-  Get the type OID by the name atom
-  """
-  @spec oid_for_name(atom() | {:array, atom()}) :: integer()
+  @spec oid_for_name(type_name | {:array, type_name}) :: oid
   def oid_for_name({:array, element_name}) do
-    case :ets.match(@ets_table_name, pg_type(name: element_name, array_oid: :"$1")) do
+    case :ets.match(@oid_table, pg_type(name: element_name, array_oid: :"$1")) do
       [[oid]] -> oid
-      _ -> raise("Unknown name {:array, #{element_name}}")
+      _ -> raise "Unknown name {:array, #{inspect(element_name)}}"
     end
   end
 
   def oid_for_name(name) do
-    case :ets.match(@ets_table_name, pg_type(name: name, oid: :"$1")) do
+    case :ets.match(@oid_table, pg_type(name: name, oid: :"$1")) do
       [[oid]] -> oid
-      _ -> raise("Unknown name #{name}")
+      _ -> raise "Unknown name #{inspect(name)}"
     end
   end
 
-  @spec type_length(integer() | atom() | {:array, atom()}) :: integer()
-  def type_length(oid) when is_integer(oid) do
-    case :ets.lookup(@ets_table_name, oid) do
-      [pg_type(length: len)] -> len
-      _ -> raise("Unknown OID #{oid}")
-    end
-  end
-
-  def type_length({:array, _}), do: raise("Can't get type length of an array type")
-
-  def type_length(name) do
-    case :ets.match(@ets_table_name, pg_type(name: name, length: :"$1")) do
-      [[len]] -> len
-      _ -> raise("Unknown name #{name}")
+  @spec pg_type_for_name(atom | binary) :: PgType.t()
+  def pg_type_for_name(name) do
+    case :ets.match_object(@oid_table, pg_type(name: name)) do
+      [pg_type() = type] -> type
+      _ -> raise "Unknown type for name #{inspect(name)}"
     end
   end
 
   def init(_) do
-    {:ok, table} =
-      ETS.Set.new(
-        name: @ets_table_name,
-        keypos: 2,
-        protection: :protected,
-        read_concurrency: true
-      )
-
-    ETS.Set.put(table, Electric.Postgres.OidDatabase.Defaults.get_defaults())
+    oid_table = :ets.new(@oid_table, [:named_table, keypos: 2, read_concurrency: true])
```

Comment:
```suggestion
    oid_table = :ets.new(@oid_table, [:set, :named_table, keypos: 2, read_concurrency: true])
```

if we're explicit here then we don't have to remember the default type for an ets table